tools/maketypes/index.js

```diff
@@ -0,0 +1,37 @@
+import {createReadStream} from "fs"
+import {fromMarkdown} from "mdast-util-from-markdown"
+import tangle from "*tangle*"
+
+let filename = process.argv[2]
+var input =
+	typeof filename == "string" ? createReadStream(filename) : process.stdin
+const chunks = []
+for await (let chunk of input) {
+	chunks.push(chunk)
+}
+const doc = fromMarkdown(Buffer.concat(chunks))
+const code = tangle(doc).replace(
+	/^(interface|type)/gm,
+	term => `export ${term}`
+)
+
+// in the full tree, externals are mandatory
+const full = code.replace(/^(\s+)external (.+)$/gm, "$1$2")
+// in the transit tree, externals must not be present
+const transit = code.replace(/^\s+external (.+:).+$/gm, "")
```

Comment:
this is the entire logic for how the two namespaces are generated. remove the `external` modifier for the full tree, remove the entire property for the transit tree